lib/Echidna/Transaction.hs

```diff
@@ -224,3 +229,36 @@ setupTx (Tx c s r g gp v (t, b)) = liftSH . sequence_ $
       SolCalldata cd  -> loadContract r >> state . calldata .= cd
     encode (n, vs) = abiCalldata
       (encodeSig (n, abiValueType <$> vs)) $ V.fromList vs
+
+saveTxs :: Maybe FilePath -> [[Tx]] -> IO ()
+saveTxs (Just d) txs = mapM_ (\v -> do let fn = d ++ "/" ++ (show . hash . show) v ++ ".txt"
+                                       b <- doesFileExist fn
+                                       unless b $ encodeFile fn (sv v)
+                             ) txs
+                             where sv = toJSON
+saveTxs Nothing  _   = return ()
+
+listDirectory :: FilePath -> IO [FilePath]
+listDirectory path = Prelude.filter f <$> getDirectoryContents path
+  where f filename = filename /= "." && filename /= ".."
+
+withCurrentDirectory :: FilePath  -- ^ Directory to execute in
+                     -> IO a      -- ^ Action to be executed
+                     -> IO a
+withCurrentDirectory dir action =
+  bracket getCurrentDirectory setCurrentDirectory $ \_ -> do
+    setCurrentDirectory dir
+    action
+
+loadTxs :: Maybe FilePath -> IO [[Tx]]
+loadTxs (Just d) = do 
+  fs <- listDirectory d
+  xs <- mapM makeRelativeToCurrentDirectory fs
+  mtxs <- withCurrentDirectory d (mapM readCall xs)
+  let txs = catMaybes mtxs
+  putStrLn ("Loaded total of " ++ show (length xs) ++ " transactions from " ++ d)
```

Comment:
I was using `xs`, because the deserialization from JSON could fail (if the files are corrupted or invalid), so we are not 100% sure they are transactions.

Reply:
right, but `xs` is pre-deserialization so it represents everything before `readCall`. the successfully loaded txs are `txs` so i don't think that's wrong?

also, we can probably collapse both `mapM`s and the `let` into a single bind if we no longer need `xs`
```haskell
txs <- catMaybes <$> withCurrentDirectory d ...
```

Reply:
I simplified this code a little to avoid `xs`. Take a look to it.